rdiodj/static/js/rdiodj.js

```diff
@@ -183,21 +183,83 @@ app.SkipList = Backbone.Firebase.Collection.extend({
   model: chat.User,
   firebase: app.roomUrl + '/skippers',
 
-  voteToSkip: function() {
-    if (!this.contains(chat.currentUser)) {
-      this.add(chat.currentUser);
-      chat.sendMessage("Voted to skip");
-    }
+  containsUser: function(user) {
+    keys = this.map(function(user){return user.get('key')});
+    return keys.indexOf(user.get('key')) !== -1;
   },
 
-  //HACK TODO there has got to be a better way to do this
-  clear: function() {
-    while (this.length > 0) {
-      this.pop();
+  voteToSkip: function() {
+    if (!this.containsUser(chat.currentUser)) {
+      this.add(chat.currentUser, {
+        success: function() {
+          chat.sendMessage("Voted to skip");
+        }
+      });
     }
   }
 });
 
+app.SkipButton = Backbone.View.extend({
+    el: "#skip-button",
+
+    events: {
+        'click': '_clickSkip'
+    },
+
+    _clickSkip: function() {
+        app.skipList.voteToSkip();
+    }
+});
+
+app.FavoriteButton = Backbone.View.extend({
+    el: "#favorite-button",
+
+    events: {
+        'click': '_clickFavorite'
+    },
+
+    initialize: function(isFavorited) {
```

Comment:
I did some preliminary testing and it seems to work fine. The button has two states based on whether the current track is favorited or not (different icon and clicking does unfavorite / favorite action depending) and the parameter specifies that.